torchvision/prototype/datasets/_builtin/dtd.py

```diff
@@ -14,30 +13,55 @@
     INFINITE_BUFFER_SIZE,
     hint_sharding,
     path_comparator,
+    BUILTIN_DIR,
     getitem,
     hint_shuffling,
 )
 from torchvision.prototype.features import Label, EncodedImage
 
+from .._api import register_dataset, register_info
+
+
+NAME = "dtd"
+
 
 class DTDDemux(enum.IntEnum):
     SPLIT = 0
     JOINT_CATEGORIES = 1
     IMAGES = 2
 
 
-class DTD(Dataset):
-    def _make_info(self) -> DatasetInfo:
-        return DatasetInfo(
-            "dtd",
-            homepage="https://www.robots.ox.ac.uk/~vgg/data/dtd/",
-            valid_options=dict(
-                split=("train", "test", "val"),
-                fold=tuple(str(fold) for fold in range(1, 11)),
-            ),
-        )
+@register_info(NAME)
+def _info() -> Dict[str, Any]:
+    categories = DatasetInfo.read_categories_file(BUILTIN_DIR / f"{NAME}.categories")
+    categories = [c[0] for c in categories]
+    return dict(categories=categories)
+
+
+@register_dataset(NAME)
+class DTD(Dataset2):
+    """DTD Dataset.
+        homepage="https://www.robots.ox.ac.uk/~vgg/data/dtd/",
+    """
+    def __init__(
+        self,
+        root: Union[str, pathlib.Path],
+        *,
+        split: str = "train",
+        fold: int = 1,
+        skip_validation_check: bool = False,
+    ) -> None:
+        self._split = self._verify_str_arg(split, "split", {"train", "val", "test"})
+
+        if not 1 <= fold <= 10:
```

Comment:
I always forget the precedence for this two-way comparison. If you don't mind, I would feel better with extra parentheses

```suggestion
        if not (1 <= fold <= 10):
```